### ISSUE TITLE
sky stop output can be misleading 

### ISSUE THREAD

**Issue by infwinston** [skypilot-org/skypilot#1972]
`sky stop` returned `Stopping cluster eval...done.` even after it failed to stop a VM.

```
E 05-19 18:14:57 cloud_vm_ray_backend.py:3136] googleapiclient.errors.HttpError: <HttpError 400 when requesting https://compute.googleapis.com/compute/v1/projects/alpa-373720/zones/us-east4-c/instances/ray-eval-head-d4fb5427-compute/stop?alt=json returned "VM has a Local SSD attached but an undefined value for `discard-local-ssd`. If using gcloud, please add `--discard-local-ssd=false` or `--discard-local-ssd=true` to your command.". Details: "[{'message': 'VM has a Local SSD attached but an undefined value for `discard-local-ssd`. If using gcloud, please add `--discard-local-ssd=false` or `--discard-local-ssd=true` to your command.', 'domain': 'global', 'reason': 'badRequest'}]">
E 05-19 18:14:57 cloud_vm_ray_backend.py:3136]
Stopping cluster eval...done.
  To restart the cluster, run: sky start eval
Stopping 1 cluster ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
```

The reason is GCP doesn't allow stopping a VM with a local ssd attached.

![image](https://github.com/skypilot-org/skypilot/assets/1206058/bf9d0e78-621a-4ab7-8366-edc7a35b6b0f)

Should we catch the error return by Ray and output message like failed to stop the cluster?

**Comment by concretevitamin** [comment]
+1, this happens for identity mismatched clusters too.

**Comment by Michaelvll** [comment]
This should be fixed by #1595

**Comment by Michaelvll** [comment]
This issue should be fixed by #1595 and #2587.
